ztoc/ztoc_marshaler_09.go

```diff
@@ -0,0 +1,209 @@
+/*
```

Comment:
This file has the functions that used to be called `flatbufToZtoc` and `ztocToFlatbuffer`. The only change is all the methods are suffixed with `09` now.

Reply:
With the above change I mentioned early, we also only need a single `ztoc_marshaller.go` that has:

- `ztocToFlatbufZtoc`
- `flatbufZtocToZtoc`
- `flatbufZtoc09ToZtoc`

(3 instead of 4 is because we really don't need a `ztocToFlatbuffer09`, since we don't want to produce old flatbuf anymore; we need `flatbufToZtoc09` because we need to be able to unmarsha old data to a `Ztoc` that without buildertool).

Reply:
I think we do want `ztocToFlatbuffer09` in the short term because we need to give some time for snapshotters to update before we produce ztoc 1.0 by default and in the long term because we're working with content addressable storage so it's important that roundtripping bytes -> ztoc -> bytes produces exactly the same bytes as before.

Reply:
This sounds like the [`BuildToolIdentifier`](https://github.com/awslabs/soci-snapshotter/blob/fb32f8751bc5d0810c595da9936ac63a8a7ff336/ztoc/ztoc.go#L39) will exist in the `Ztoc` go struct forever?

Unless we have a `Ztoc09` struct and `Ztoc` struct in our go code, and the `Ztoc` will be moved to the standalone `Ztoc` library in the future while `Ztoc09` stays in SOCI. Because otherwise it doesn't make sense in the standalone `Ztoc` lib the `Ztoc` struct has a deprecated `BuildToolIdentifier` field.

Reply:
Good point. To get around this, I think we can just hard code the value or mark the field deprecated. Hardcoding should mean we continue to produce the same indices forever. Deprecating means everything gets new digests after the change (but at least the old ones still work).

Reply:
Do you mean we remove the field in flatbuf, but keep the field in our `Ztoc` struct (and comment that this field is deprecated; or hardcode this value)?

I think to really "have a clean v1.0 that doesn't have deprecated fields", we need to remove the field in both flatbuf (to work around we make a 09 flatbuf copy now) AND `Ztoc` struct (e.g., this field also shouldn't appear in `Ztoc` struct, either as deprecated or hardcoded, in the standalone ztoc lib.)

Reply:
I mean we have 1 `Ztoc` struct without the field and we hardcode the value that goes into the flatbuffer in `ztocToFlatbuffer09`. Thinking a little deeper, we have to hardcode it to what we currently have (which is `”AWS SOCI CLI v0.1”`) to make sure that re-serializing produces the same digests.